decoder.go

```diff
@@ -6,6 +6,20 @@ import (
 	"io"
 )
 
+const (
+	// MaximumRecordLength defines the maximum length in bytes, acceptable for records while decoding.
+	// This maximum prevents from excessive memory allocation for decoding.
+	// The value is derived from MAX_PKT_SIZ 65536 in sflow reference implementation
+	// https://github.com/sflow/sflowtool/blob/bd3df6e11bdf8261a42734c619abfe8b46e1202f/src/sflowtool.c#L4313
+	MaximumRecordLength = 65536
+
+	// MaximumHeaderLength defines the maximum length in bytes, acceptable for packet flow samples while decoding.
+	// This maximum prevents from excessive memory allocation for decoding.
+	// The value is derived from INM_MAX_HEADER_SIZE 256 in sflow reference implementation
+	// https://github.com/sflow/sflowtool/blob/bd3df6e11bdf8261a42734c619abfe8b46e1202f/src/sflowtool.h#L28
+	MaximumHeaderLength = 256
```

Comment:
I agree, let's not make it too high :). I'll let you choose either 1300 or 1500.